resolwe/process/models.py

```diff
@@ -331,6 +331,38 @@ def filter(cls, **filters: Dict[str, Any]) -> List["Model"]:
             models.append(model)
         return models
 
+    @classmethod
+    def iterate(
+        cls, sort: Optional[List[str]] = None, **filters: Dict[str, Any]
+    ) -> List["Model"]:
+        """Return a iterator over all objects that fit criteria."""
+        # Make sure attributes have 'id' in the first place.
+        attributes = filters.pop("__fields", None)
+        attributes = attributes or cls._filter_response_fields
+        attributes = ["id"] + [
+            attribute for attribute in attributes if attribute != "id"
+        ]
+        sort = sort or ["id"]
+        offset = 0
+        final_iteration = False
+        while not final_iteration:
+            results = communicator.iterate_objects(
+                cls._app_name, cls._model_name, filters, sort, attributes, offset
+            )
+            assert results["starting_offset"] == offset, (
+                f"Offset mismatch while iterating, got {results['starting_offset']} "
+                f"expected {offset}."
+            )
+            objects = results["objects"]
+            offset = results["starting_offset"] + len(objects)
```

Comment:
Update the offset at the end of the method, as otherwise, one may accidentally use the wrong one.